### PR TITLE
scan: use real code from server

### LINKED ISSUE #340: Implement CodeHook
Currently, CodeHook has a mocked code. Please use Axios to query real data from an endpoint.

### PR DESCRIPTION
**Pull request by evilpeach**
Fixes #340

### CODE REVIEW

File: scan/src/rpc/CodeHook.re [hunk 1]
```diff
@@ -1,84 +1,27 @@
...
-    #[test]
-    fn test_end_to_end_from_local_env() {
-        // Run with local environment
-        let data = Data::build_from_local_env().unwrap();
-        println!("Current BTC price (times 100) is {}", execute(vec![data]));
-    }
-}|f},
-    },
-  ];
+let getCode = (codeHash: Hash.t) => {
```
Comment (sorawit) [inline scan/src/rpc/CodeHook.re:18]: What happens if the code can't be found? We need to make sure we handle that gracefully.

File: scan/src/utils/Hash.re [hunk 2]
```diff
@@ -9,6 +9,9 @@ let toBase64 =
   fun
   | Hash(hash) => hash->JsBuffer.hexToBase64;
 
-let toHex = (~with0x=false) =>
+let toHex = (~with0x=false, ~upper=false) =>
```
Comment (sorawit) [inline scan/src/utils/Hash.re:12]: Please add unit tests to `Hash_test.re`. Consult @prin-r if you have questions.

**Review by sorawit — CHANGES_REQUESTED** [review]

File: scan/src/rpc/CodeHook.re (hunk 1 above)
Comment (evilpeach) [reply scan/src/rpc/CodeHook.re:18]: In the case of any error, the ScriptCode Component will render "Code Not Found".

**Review by sorawit — DISMISSED** [review]
lgtm

**Review by prin-r — APPROVED** [review]

**Review by sorawit — APPROVED** [review]

**Pull request merged**